### DISCUSSION TITLE
7.2.1 not removing drm from IBS ebook

### DISCUSSION THREAD

**Discussion by Loob22** [apprenticeharper/DeDRM_tools#2032] [Q&A]
hi,
Lorenzo
windows 10
bought a ebook from IBS
the 7.2.1 version is not removing the drm
i used adobe to visualise the file before using the dedrm tool, if this can be a problem

log:
calibre Log di debug
calibre 5.38  embedded-python: True is64bit: False
Windows-10-10.0.19041 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: it
Successfully initialized third party plugins: DeDRM (7, 2, 1)
calibre 5.38  embedded-python: True is64bit: False
Windows-10-10.0.19041 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: it
Successfully initialized third party plugins: DeDRM (7, 2, 1)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 120.0 x 120.0
physicalDpi: 141.7674418604651 x 141.4020618556701
Using calibre Qt style: True
[0.00] Starting up...
[0.02] Showing splash screen...
[0.50] splash screen shown
[0.50] Initializing db...
[0.50] db initialized
[0.50] Constructing main UI...
[1.02] main UI initialized...
[1.02] Hiding splash screen
Starting QuickView
DeDRM v7.2.1: Trying to decrypt Meccatronica_Azionamenti_Elettrici_ed_Ol (1).pdf
DeDRM v7.2.1: Meccatronica_Azionamenti_Elettrici_ed_Ol (1).pdf is a PDF ebook
DeDRM v7.2.1: Looking for new default Adobe Digital Editions Keys after 0.0 seconds
Found 1 keys
DeDRM v7.2.1: Trying a new default key
DeDRM v7.2.1: Exception when decrypting after 0.0 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 487, in PDFDecrypt
  File "calibre_plugins.dedrm.ineptpdf", line 2178, in decryptBook
  File "calibre_plugins.dedrm.ineptpdf", line 1984, in __init__
  File "calibre_plugins.dedrm.ineptpdf", line 1460, in initialize
calibre_plugins.dedrm.ineptpdf.PDFEncryptionError: Document is not encrypted.
DeDRM v7.2.1: Failed to decrypt with new default key after 0.0 seconds
DeDRM v7.2.1: Ultimately failed to decrypt after 0.0 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre\customize\ui.py", line 175, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 640, in run
  File "calibre_plugins.dedrm.__init__", line 515, in PDFDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v7.2.1: Ultimately failed to decrypt after 0.0 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Exception ignored in: <function NativeFunction.__del__ at 0x07F156A0>
Traceback (most recent call last):
  File "calibre_plugins.dedrm.adobekey", line 268, in __del__
TypeError: 'NoneType' object is not callable
Exception ignored in: <function NativeFunction.__del__ at 0x07F156A0>
Traceback (most recent call last):
  File "calibre_plugins.dedrm.adobekey", line 268, in __del__
TypeError: 'NoneType' object is not callable
[7.72] splash screen hidden
[7.72] Started up in 7.72 seconds with 1 books
stdout+stderr from file dialog helper: [b'', b'']
piped data from file dialog helper: [b'\xa8\x7fw\x14\xde\xc9\x12\x98\xa3\x9f\x8fR\x9b\xc7\x88\xd0Z\x0f\xaf\xc2\x13\\\xb0\x10 \x87odf\x04\x19Z', b'C:\\Users\\loreb\\Downloads\\Meccatronica_Azionamenti_Elettrici_ed_Ol (1).pdf']
Added () to db in: 0.1
Added 1 books in 0.6 seconds

**Comment by ElleKayEm** [comment]
Can you read this with Adobe Digital Editions authorized with your Adobe ID?

You might want to try noDRM's DeDRM: https://github.com/noDRM/DeDRM_tools

**Reply by Loob22** [reply]
yes, but unfortunately i deleted the book from ADE library and now i can no more.

**Reply by ElleKayEm** [reply]
You can't download a fresh copy?

**Reply by Loob22** [reply]
if i download a fresh copy and try to visualise it Adobe only let me see the first page

**Reply by ElleKayEm** [reply]
I don't know.  DeDRM depends on you being able to read the original file with your account info in the original program.  Give noDRM's DeDRM a try anyway.

**Reply by Loob22** [reply]
> You can't download a fresh copy?

hi, now i can see the full book on adobe digital edition, but the plugin still don't work on calibre

**Reply by ElleKayEm** [reply]
ADE is on the same computer and user as calibre with DeDRM?  Have you tried DeDRM 10.0.2 yet?

**Reply by Loob22** [reply]
hi, yes they are on the same computer.
no, i used 7.2.1

**Reply by Loob22** [reply]
where can i download it? i thught 7.2.1 was the latest version

**Reply by ElleKayEm** [reply]
The link I gave above.

**Reply by Loob22** [reply]
yeah i downloaded it, a .zip called dedrm_tools_master, but there are no dedrm_plugins.zip to use inside of it

**Reply by ElleKayEm** [reply]
Download the latest release here:  https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.2

**Reply by Loob22** [reply]
it worked, thank you so much. now i can see a book for an important exam at uni
